packages/xml-views-validation/test/validators/attributes/use-of-depracated-attribute-spec.ts

```diff
@@ -0,0 +1,164 @@
+import { partial, find } from "lodash";
+import { expect } from "chai";
+import { parse, DocumentCstNode } from "@xml-tools/parser";
+import { buildAst } from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import {
+  generateModel,
+  expectExists,
+} from "@ui5-language-assistant/test-utils";
+import {
+  assertNoIssues as assertNoIssuesBase,
+  assertSingleIssue as assertSingleIssueBase,
+} from "../../test-utils";
+import { validateUseOfDeprecatedAttribute } from "../../../src/validators/attributes/use-of-depracated-attribute";
+import {
+  buildDeprecatedIssueMessage,
+  DeprecatedUI5Symbol,
+} from "../../../src/utils/deprecated-message-builder";
+
+describe("the use of deprecated attribute validation", () => {
+  let ui5SemanticModel: UI5SemanticModel;
+
+  before(async () => {
+    ui5SemanticModel = await generateModel({ version: "1.74.0" });
+  });
+
+  context("true positive scenarios", () => {
+    let assertSingleIssue: (xmlSnippet: string, message: string) => void;
+    before(() => {
+      assertSingleIssue = partial(
+        assertSingleIssueBase,
+        ui5SemanticModel,
+        {
+          attribute: [validateUseOfDeprecatedAttribute],
+        },
+        "UseOfDeprecatedAttribute",
+        "warn"
+      );
+    });
+
+    it("will detect usage of a deprecated attribute property", () => {
+      const pageClass = ui5SemanticModel.classes["sap.m.Page"];
+      const navButtonTextProperty = find(
+        pageClass.properties,
+        (_) => _.name === "navButtonText"
+      );
+
+      assertSingleIssue(
+        `<mvc:View xmlns:m="sap.m" 
+          xmlns:mvc="sap.ui.core.mvc">
+          <m:Page 🢂navButtonText🢀="">
+          </m:Page>
+        </mvc:View>`,
+        buildDeprecatedIssueMessage({
+          symbol: navButtonTextProperty as DeprecatedUI5Symbol,
+          model: ui5SemanticModel,
+        })
+      );
+    });
+
+    it("will detect usage of a deprecated attribute event", () => {
+      const appClass = ui5SemanticModel.classes["sap.m.App"];
+      const orientationChangeEvent = find(
+        appClass.events,
+        (_) => _.name === "orientationChange"
+      );
+
+      assertSingleIssue(
+        `<mvc:View xmlns:m="sap.m" 
+          xmlns:mvc="sap.ui.core.mvc">
+          <m:App 🢂orientationChange🢀="">
+          </m:App>
+        </mvc:View>`,
+        buildDeprecatedIssueMessage({
+          symbol: orientationChangeEvent as DeprecatedUI5Symbol,
```

Comment:
if you use `assertExists` or `expectExists` w(e.g in line 67)
would it remove the need for the cast?

This comments refers to all uses of this pattern in this file.